src/controllers/employees.js

```diff
@@ -1,156 +1,60 @@
-    if (err) {
-      res.send('Cannot update employee');
-    } else {
-      res.send('Employee updated successfully');
+    return res.status(200).json({
+      message: `Employee with id ${id} deleted`,
+      data: result,
+      error: false,
+    });
+  } catch (error) {
+    let status = 404;
+    if (!error.message.includes('Employee not found')) {
+      status = 404;
     }
-  });
-});
-
-router.get('/get', (req, res) => {
-  res.send(employees);
-});
-
-router.get('/get/:id', (req, res) => {
-  const employeeId = req.params.id;
-  const findEmployee = employees.find((emp) => emp.id === employeeId);
-  if (findEmployee) {
-    res.send(findEmployee);
-  } else {
-    res.send('User not found.');
-  }
-});
-
-router.get('/getTask/:task', (req, res) => {
-  const taskParam = req.params.task;
-  const filteredTask = employees.filter((t) => t.task === taskParam);
-  if (filteredTask.length > 0) {
-    res.send(filteredTask);
-  } else {
-    res.send('Task not found');
-  }
-});
-
-router.get('/getName/:firstName', (req, res) => {
-  const nameParam = req.params.firstName;
-  const filteredName = employees.filter((fn) => fn.firstName === nameParam);
-  if (filteredName.length > 0) {
-    res.send(filteredName);
-  } else {
-    res.send('First Name not found');
-  }
-});
-
-router.get('/getLastName/:lastName', (req, res) => {
-  const lastNameParam = req.params.lastName;
-  const filteredLastName = employees.filter((ln) => ln.lastName === lastNameParam);
-  if (filteredLastName.length > 0) {
-    res.send(filteredLastName);
-  } else {
-    res.send('Last Name not found');
+    return res.status(status).json({
+      message: error.toString(),
+      data: undefined,
+      error: true,
+    });
   }
-});
-
-router.get('/getProject/:projectId', (req, res) => {
-  const EmployeeProjectId = req.params.projectId;
-  const filteredProject = [];
-  employees.forEach((e) => {
-    if (e.projects.find((p) => p.projectId === EmployeeProjectId)) {
-      filteredProject.push(e);
+};
+
+const updateEmployees = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const result = await Employees.findByIdAndUpdate(
+      req.params.id,
+      req.body,
+      { new: true },
+    );
+    if (!result) {
+      throw new Error(`Employee ${req.query.id} not found`);
     }
-  });
-  if (filteredProject.length > 0) {
-    res.send(filteredProject);
-  } else {
-    res.send('Project not found');
-  }
-});
-
-router.get('/getRole/:role', (req, res) => {
-  const roleParam = req.params.role;
-  const filteredRole = [];
-  employees.forEach((e) => {
-    if (e.projects.find((r) => r.role === roleParam)) {
-      filteredRole.push(e);
+    return res.status(200).json({
+      message: `Employee with id ${id} edited`,
+      date: result,
+      error: false,
+    });
+  } catch (error) {
+    let status = 400;
+    if (!error.message.includes('Employee not found')) {
```

Comment:
The error message will never match. Please try to understand how this logic works!

```suggestion
    if (!error.message.includes(`Employee ${req.query.id} not found`)) {
```